members-service/updateMember.js

```diff
@@ -17,21 +17,28 @@ const logger = new OperationsLogger({
 });
 
 export const handlerFunc = (event: any, context: any, callback: any) => {
+  let payload;
+  if (event.body) payload = JSON.parse(event.body);
+  else payload = {};
+
   const parameters = {
     ...event.pathParameters,
-    ...JSON.parse(event.body),
+    ...payload,
   };
-  const { id, ...data } = parameters;
+
   return validateRequest(UPDATE_MEMBER_SCHEMA, parameters).then(
     params => {
-      update(id, data)
+      update(parameters.id, parameters.member)
         .then(result => {
-          logger.log({
+          return logger.log({
```

Comment:
You're now returning the logger result rather than the AK result. Is that on purpose? If you're feeling like refactoring, I think this method would be a bit more readable with chained then blocks and a catch block.

Reply:
I'm returning the logger promise, because given the other case, if the logger throws an exception, then this method will still return a successful response instead of an error. 

But I think you just spotted the bug I was experiencing, since the "result" returned from the AK client is the one that the `response` method needs.

Reply:
@rodrei @Tuuleh nice job spotting this issue there! I didn't think of it at first when I wrote it (swallowing the `logger.log` exceptions since i'm not handling that promise's reject scenario) Does this mean we're now back to not handling the logger exceptions (if they happen)?

I think @rodrei's first refactor was on the right track (handling the logger's result) but the `Promise` syntax seems unsuitable for this -- we'd have to wrap that block in another promise, which has that undesired cascading look. I guess with `async / await` it might be clearer:

```js
export const handlerFunc = async (event: any, context: any, callback: any) => {
  const payload = event.body ? JSON.parse(event.body) : {};
  const parameters = {
    ...event.pathParameters,
    ...payload,
  };

  // validate the request
  // if the request is invalid, return badRequest
  try {
    await validateRequest(UPDATE_MEMBER_SCHEMA, parameters);
  } catch (e) {
    return callback(null, badRequest({ cors: true, body: e }));
  }

  // update and log
  // if either fails, catch and return the error
  // on success, return update response
  try {
    const result = await update(parameters.id, parameters.member);
    await logger.log({
      event: 'MEMBER:UPDATE',
      data: {
        akId: parameters.id,
        email: parameters.email,
        params: parameters.member,
      },
      status: {
        actionkit: 'SUCCESS',
        champaign: 'PENDING',
      },
    });
    return callback(null, response(result));
  } catch (e) {
    return callback(null, response({ cors: true, body: e }));
  }
};
```

The other question I have now is, what do we want to do when the update happened on AK but the dynamodb record was not saved? Would that ever happen though? I guess local testing is the only environment that we'll encounter that, since we don't really have a local dynamodb.